design/ceph-config-updates.md

```diff
@@ -0,0 +1,235 @@
+# Updating Rook's Ceph configuration strategy (Jan 2019)
+**Targeted for v1.0-v1.1**
+
+## Background
+
+Starting with Ceph Mimic, Ceph is able to store the vast majority of config options for all daemons
+in the Ceph mons' key-value store. This ability to centrally manage 99% of its configuration
+is Ceph's preferred way of managing config. This conveniently allows Ceph to fit better into the
+containerized application space than before.
+
+However, for it's own backwards compatibility, Ceph options set in the config file or via command
+line flags will override the centrally-configured settings. To make the most of this functionality
+within Ceph, it will be necessary to limit the configuration options specified by Rook in either
+config files or on the command line to a minimum.
+
+
+## Proposed new design
+
+The end goal of this work will be to allow Ceph to centrally manage its own configuration as much as
+possible. Or in other terms, Rook will specify the barest minimum of configuration options in the
+config file or on the command line with a priority on cleaning the config file.
+
+Rook currently offers the option of a config override, and this functionality can be kept for
+backwards compatibility for existing users; however, we should note that it is advisable not to do
```

Comment:
Can we remove support for this? In the release notes we could point out that they need to apply their settings differently and give examples of how to convert the override configmap to the new settings.

Reply:
To record an offline discussion, it could be removed, but this is the simplest way to achieve the same functionality. Adding extra commandline flags to ceph daemons via CRD would require a CRD update. And if Rook no longer needs to create a config file, the user-specified config override can be the config file in Ceph daemon containers when it has been specified.